src/extension/serializer.ts

```diff
@@ -1009,16 +1047,395 @@ export class GrpcSerializer extends SerializerBase {
     this.serverReadyListener?.dispose()
     super.dispose()
   }
+}
 
-  public getMaskedCache(cacheId: string): Promise<Uint8Array> | undefined {
-    return this.maskedCache.get(cacheId)
+export class ConnectSerializer extends GrpcSerializerBase {
+  log: ReturnType<typeof getLogger>
+  client: PromiseClient<typeof ParserService>
+  serializerServiceUrl: string
+  protected ready: ReadyPromise
+
+  private serverReadyListener: Disposable | undefined
+
+  constructor(
+    protected context: ExtensionContext,
+    serializerAddress: string,
+    kernel: Kernel,
+  ) {
+    super(context, kernel)
+    this.log = getLogger('ConnectSerializer')
+    this.serializerServiceUrl = serializerAddress
+    this.client = this.createSerializerClient()
+    this.ready = new Promise((resolve) => {
+      resolve()
+    })
   }
 
-  public getPlainCache(cacheId: string): Promise<Uint8Array> | undefined {
-    return this.plainCache.get(cacheId)
+  createSerializerClient = () => {
+    this.log.info(`Serializer: Client pointed to: ${this.serializerServiceUrl}`)
+    // Server options:
+    // (1) pre-existing, started internally by this extension (assuming local execution so it has permissions to start)
+    // (2) pre-existing, started externally, presumably at the this.serializerServiceUrl address
+    //   In both cases we need a key & cert (assuming tls) which may reside in:
+    //   (a) this project: '/path/to/projectRoot/tls'
+    //   (b) the runme binary server default: '~/.config/runme/tls'
+    //   (c) other, manually configured
+
+    // I will assume 2a here
```

Comment:
How does the gRPC serializer determine the appropriate location for the cert?

Reply:
(the short version): very close to the same as I do, using the path the extension resides plus a `'/tls'` string.  
(the long version): The GrpcSerializer gets it from the the transport which is constructed in the kernel which gets it from the constructor arg, which gets it from vscode extension base path, which has an '/tls' string added via the `getTLSDir` configuration function. I can't, however, access the kernel methods because they are protected & private. I do have access to the same vscode context, and configuration function, so I could do the following replacement if desirable within the `ConnectSerializer.createSerializerClient` method. 
`const tlsPath = path.normalize(__dirname + '/../tls')`
replace with
`const tlsPath = getTLSDir(Uri.parse(this.context.extensionPath))`
of course this assumes you import `getTLSDir` from the configuration file

Reply:
Since I've already done the legwork to suss out the aforementioned call, I'll commit it as an update to this PR soon.

Reply:
Should we move transport construction into the kernel? Should we change the kernel API to expose some other information like the tlsDir?

What's the cleanest most robust way to do this?

Reply:
the way to remove the most code is likely to expose the `getTLS` method in kernelServer.ts so the certs can be used when constructing the client. I don't know if there are security concerns in doing this, it seems very possible. If we are trying to move away from using the file system because we want to run in the browser then we are in new territory and a decision has to be made on where this tls data is to come from.